app/frontend/src/routes.ts

```diff
@@ -67,27 +67,18 @@ export const routeToEvent = (id: number, slug: string) =>
   `${eventBaseRoute}/${id}/${slug}`;
 
 const communityBaseRoute = "/community";
-export const communityRoute = `${communityBaseRoute}/:communityId/:communitySlug?`;
-export const routeToCommunity = (id: number, slug: string) =>
-  `${communityBaseRoute}/${id}/${slug}`;
-
-const communityPlacesBaseRoute = `${communityRoute}/places`;
-export const communityPlacesRoute = `${communityPlacesBaseRoute}/:communityId/:communitySlug?`;
-export const routeToCommunityPlaces = (id: number, slug: string) =>
-  `${communityPlacesBaseRoute}/${id}/${slug}`;
-const communityGuidesBaseRoute = `${communityRoute}/guides`;
-export const communityGuidesRoute = `${communityGuidesBaseRoute}/:communityId/:communitySlug?`;
-export const routeToCommunityGuides = (id: number, slug: string) =>
-  `${communityGuidesBaseRoute}/${id}/${slug}`;
-const communityGroupsBaseRoute = `${communityRoute}/groups`;
-export const communityGroupsRoute = `${communityGroupsBaseRoute}/:communityId/:communitySlug?`;
-export const routeToCommunityGroups = (id: number, slug: string) =>
-  `${communityGroupsBaseRoute}/${id}/${slug}`;
-const communityDiscussionsBaseRoute = `${communityRoute}/discussions`;
-export const communityDiscussionsRoute = `${communityDiscussionsBaseRoute}/:communityId/:communitySlug?`;
-export const routeToCommunityDiscussions = (id: number, slug: string) =>
-  `${communityDiscussionsBaseRoute}/${id}/${slug}`;
-const communityEventsBaseRoute = `${communityRoute}/events`;
-export const communityEventsRoute = `${communityEventsBaseRoute}/:communityId/:communitySlug?`;
-export const routeToCommunityEvents = (id: number, slug: string) =>
-  `${communityEventsBaseRoute}/${id}/${slug}`;
+const communityTabs = [
+  "overview",
+  "find-host",
+  "events",
+  "local-points",
+  "discussions",
+  "hangouts",
+] as const;
+export type CommunityTab = typeof communityTabs[number];
```

Comment:
`as const`, my new favourite TS magic. That let's `CommunityTab` be a union of those strings instead of just `string`!

Reply:
Ahh yeah, `as const` is probably the one exception where it's mostly fine/good to use a type assertion to help narrow the type.

Although it doesn't guarantee the array will remain constant, as it is still a type assertion so you could still do `communityTabs.push(...)` and things would break. But hopefully that'd be weird enough (as there wouldn't be a good reason to do that and then use `as const`) that if anyone tries to do that, we'd pick that up in PR reviews...

Reply:
lol, I was using the array for something but I must have refactored it out...